pygments/lexers/q.py

```diff
@@ -0,0 +1,228 @@
+                bygroups(Name.Function, Whitespace, Operator, Punctuation),
+                "functions",
+            ),
+            # Variable Names
+            (
+                r"([.]?[a-zA-Z](?:\w|[.])*)(\s*)([-.~=!@#$%^&*_+|,<>?/\\:']?:)",
+                bygroups(Name.Variable, Whitespace, Operator),
+            ),
+            # Functions
+            (r"\{", Punctuation, "functions"),
+            # Parentheses
+            (r"\(", Punctuation, "parentheses"),
+            # Brackets
+            (r"\[", Punctuation, "brackets"),
+            # Errors
+            (r"'`[a-zA-Z]*(\w|[.])*", Name.Exception),
+            # File Symbols
+            (r"`:[a-zA-Z/]*(\w|[./])*", String.Symbol),
+            # Symbols
+            (r"[`][a-zA-Z]*(\w|[.])*", String.Symbol),
+            # Numbers
+            include("numbers"),
+            # Variable Names
+            (r"[a-zA-Z](\w|[.])*", Name),
+            # Operators
+            (r"[-=+*#$%@!~^&:.,<>'\\|/?_]", Operator),
+            # Punctuation
+            (r"[;]", Punctuation),
+        ],
+        "functions": [
+            include("root"),
+            (r"\}", Punctuation, "#pop"),
+        ],
+        "parentheses": [
+            include("root"),
+            (r"\)", Punctuation, "#pop"),
+        ],
+        "brackets": [
+            include("root"),
+            (r"\]", Punctuation, "#pop"),
+        ],
+        "numbers": [
+            (r"[01]+b", Number.Bin),
+            (r"0[nNwW][cefghijmndzuvtp]?", Number),  # nulls/infinities
+            (
+                (
+                    r"(?:[0-9]{4}[.][0-9]{2}[.][0-9]{2}|[0-9]+)"
+                    "D(?:[0-9](?:[0-9](?:[:][0-9]{2}"
+                    "(?:[:][0-9]{2}(?:[.][0-9]*)?)?)?)?)?"
+                ),
+                Literal.Date,  # timestamp
+            ),
+            (
+                (
+                    r"[0-9]{4}[.][0-9]{2}"
+                    "(?:m|[.][0-9]{2}(?:T(?:[0-9]{2}:[0-9]{2}"
+                    "(?:[:][0-9]{2}(?:[.][0-9]*)?)?)?)?)"
+                ),
+                Literal.Date,  # datetime
+            ),
+            (
+                (r"[0-9]{2}:[0-9]{2}" "(?:[:][0-9]{2}" "(?:[.][0-9]{1,3})?)?"),
+                Literal.Date,  # time
+            ),
+            (
+                r"[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}",
+                Number.Hex,  # GUID
+            ),
+            (r"0x[0-9a-fA-F]+", Number.Hex),
+            (r"[0-9]+[jnp]", Number.Integer.Long),
+            (r"[0-9]+[ihtuv]", Number.Integer),
+            (r"([0-9]*[.]?[0-9]+|[0-9]+[.]?[0-9]*)[eE][+-]?[0-9]+[ef]?", Number.Float),
+            (r"([0-9]*[.][0-9]+|[0-9]+[.][0-9]*)[ef]?", Number.Float),
+            (r"[0-9]+[ef]", Number.Float),
+            (r"[0-9]+[c]", Number),
+            (r"[0-9]+", Number.Integer.Long),
+        ],
+        "comments": [
+            (r"[^\\]", Comment.Multiline),
```

Comment:
This is a known antipattern: it generates one token per character, which makes the raw token output (such as in the regression test) much harder to read, and can slow down highlighting too. Please add a `+` quantifier.

Another way to do it would be `^/\s*\n(.|\n)*?^\\` to match a multiline comment and do without a dedicated state altogether.

Reply:
thank you for the suggestion.  i hated the regression test output as well!  this is a result of following the [pygment documentation](https://pygments.org/docs/lexerdevelopment/#changing-states).  specifically:

```python
    tokens = {
        'root': [
            (r'[^/]+', Text),
            (r'/\*', Comment.Multiline, 'comment'),
            (r'//.*?$', Comment.Singleline),
            (r'/', Text)
        ],
        'comment': [
            (r'[^*/]', Comment.Multiline),
            (r'/\*', Comment.Multiline, '#push'),
            (r'\*/', Comment.Multiline, '#pop'),
            (r'[*/]', Comment.Multiline)
        ]
    }
```